### DISCUSSION TITLE
uv__tty_console_resize_*_thread

### DISCUSSION THREAD

**Discussion by martinzzrrxx** [libuv/libuv#4485] [Q&A]
I'm using libuv in my server library. When I debug my server library, I founded there are always two background threads are running:

`uv__tty_console_resize_watcher_thread`
`uv__tty_console_resize_message_loop_thread`

I want to figure out why they are started by default? Can I disable them in my lib, and what's the affect?

Thanks!

**Comment by saghul** [comment]
They are both used for dealing with the console when tty handles are used.

If you are not using that then they will be idle. Is that a problem?

**Reply by martinzzrrxx** [reply]
For me, it will better if I can disable them, so they don't consume extra two threads althogh they're always in idle state.

**Reply by vtjnash** [reply]
It looks like one of the two threads is a bit extraneous, since libuv could PostQueuedCompletionStatus the SIGWINCH to the various IOCP a bit more directly than it does currently. Not really an issue, unless someone feels inclined to micro optimize this slightly more.

**Reply by saghul** [reply]
Yeah and I think the threads could likely be lazily initialized when a tty handle is created (just once) right?

**Reply by vtjnash** [reply]
That would be a breaking change. But it could be lazily initialized when a SIGWINCH handler is installed, since its only purpose is to feed that signal correctly

**Reply by vtjnash** [reply]
https://github.com/libuv/libuv/pull/4488

**Reply by saghul** [reply]
> That would be a breaking change. But it could be lazily initialized when a SIGWINCH handler is installed, since its only purpose is to feed that signal correctly

Why would it be a breaking change? Isn't it transparent to the user?

**Reply by vtjnash** [reply]
It is a signal handler dispatcher, unrelated to the presence of tty handles (sorry if that wasn't super clear in my previous response)

**Reply by martinzzrrxx** [reply]
@vtjnash Sorry, I didn't understand the issue(I did read the issue carefull, but still don't understand), is it related to my question?

**Reply by saghul** [reply]
> It is a signal handler dispatcher, unrelated to the presence of tty handles (sorry if that wasn't super clear in my previous response)

Ah right, I read it too quickly 😅
